commands/channel_test.go

```diff
@@ -129,3 +129,150 @@ func (s *MmctlUnitTestSuite) TestSearchChannelCmdF() {
+
+		cmd := &cobra.Command{}
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(), "Name is required")
+		s.Require().Len(printer.GetLines(), 0)
+
+	})
+
+	s.Run("Create channel with a name but no Display Name returns an error", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(),"Display Name is required")
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Create channel with a Name but no Display Name returns an error", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(),"Name is required")
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Create channel with a Name and Display Name but Team returns an error", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(), "Team is required")
+		s.Require().Len(printer.GetLines(), 0)
+
+	})
+
+	s.Run("Test creating a channel with a nonexistent team returns an error", func() {
+		printer.Clean()
+
+		teamNameArg := "mockTeamName"
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		cmd.Flags().String("team", teamNameArg, "")
+
+		s.client.
+			EXPECT().
+			GetTeam(teamNameArg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamNameArg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+
+		s.Require().Equal(err.Error(), "Unable to find team: "+ teamNameArg)
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Create open channel", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", mockChannelName, "")
+		cmd.Flags().String("display_name", mockChannelDisplayName, "")
+		cmd.Flags().String("team", mockTeam.Name, "")
+
+		mockChannel := &model.Channel{
+			Name:        mockChannelName,
+			DisplayName: mockChannelDisplayName,
+			TeamId: mockTeam.Id,
+			Type: model.CHANNEL_OPEN,
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(mockTeam.Name, "").
+			Return(mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			CreateChannel(mockChannel).
+			Return(mockChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, []string{})
+		s.Require().Nil(err)
+		s.Require().Equal(mockChannel, printer.GetLines()[0])
+		s.Require().Len(printer.GetLines(), 1)
+	})
+
+	s.Run("Create invite only channel", func() {
```

Comment:
For consistency with the naming, I think we should use `private` here:

```suggestion
	s.Run("Create private channel", func() {
```